src/main/java/com/zufar/onlinestore/cart/api/ShoppingSessionProvider.java

```diff
@@ -29,6 +33,23 @@ public ShoppingSessionDto getByUserId(final UUID userId) throws ShoppingSessionN
             log.error("The shopping session for the user with id = {} is not found.", userId);
             throw new ShoppingSessionNotFoundException(userId);
         }
+        return shoppingSessionDtoConverter.toDto(shoppingSession); //ToDo think about return type
+    }
+
+    @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.READ_COMMITTED)
+    public ShoppingSessionDto removeItemFromShoppingSession(final RemoveItemFromShoppingSessionRequest request) {
+        Integer deletedItems = shoppingSessionItemRepository.deleteShoppingSessionItemById(request.shoppingSessionItemId());
+        if (deletedItems < request.shoppingSessionItemId().size()) {
```

Comment:
Why do u need this check?! As far as I understand when you delete an item you always return item deleted id. So in that case why did u need this if check? Or if I'm wrong please rename the method because it's more likely unclear